tests/mock_crust/churn.rs

```diff
@@ -116,134 +114,76 @@ fn count_received(group: Vec<&TestNode>,
                   message_id: MessageId)
                   -> usize {
     group.iter()
-         .filter(|node| did_receive_get_request(node,
-                                                src,
-                                                dst,
-                                                data_id,
-                                                message_id))
-         .count()
+        .filter(|node| did_receive_get_request(node, src, dst, data_id, message_id))
+        .count()
 }
 
 fn quorum(group_len: usize) -> usize {
     (group_len * QUORUM - 1) / 100 + 1
 }
 
+const CHURN_ITERATIONS: usize = 100;
+
 #[test]
 fn churn() {
     let network = Network::new(None);
 
     let mut rng = network.new_rng();
     let mut nodes = create_connected_nodes(&network, 20);
 
-    for i in 0..100 {
+    for i in 0..CHURN_ITERATIONS {
         trace!("Iteration {}", i);
-        let _ = random_churn(&mut rng, &network, &mut nodes);
-        poll_and_resend(&mut nodes, &mut []);
-
-        for node in &mut nodes {
-            node.inner.clear_state();
-        }
-
-        verify_invariant_for_all_nodes(&nodes);
-    }
-}
-
-const REQUEST_DURING_CHURN_ITERATIONS: usize = 10;
-
-#[test]
-#[ignore]
-fn request_during_churn_node_to_self() {
-    let network = Network::new(None);
-    let mut rng = network.new_rng();
-
-    let mut nodes = create_connected_nodes(&network, 2 * MIN_GROUP_SIZE);
-
-    for _ in 0..REQUEST_DURING_CHURN_ITERATIONS {
-        let added_index = random_churn(&mut rng, &network, &mut nodes);
-        let index = gen_range_except(&mut rng, 0, nodes.len(), added_index);
-        let name = nodes[index].name();
-
-        let src = Authority::ManagedNode(name);
-        let dst = Authority::ManagedNode(name);
-        let data = gen_immutable_data(&mut rng, 8);
-        let data_id = data.identifier();
-        let message_id = MessageId::new();
-
-        unwrap!(nodes[index].inner.send_get_request(src, dst, data_id, message_id));
-
-        poll_and_resend(&mut nodes, &mut []);
-        assert!(did_receive_get_request(&nodes[index], src, dst, data_id, message_id));
-    }
-}
-
-#[test]
-#[ignore]
-fn request_during_churn_node_to_node() {
-    let network = Network::new(None);
-    let mut rng = network.new_rng();
-    let mut nodes = create_connected_nodes(&network, 2 * MIN_GROUP_SIZE);
-
-    for _ in 0..REQUEST_DURING_CHURN_ITERATIONS {
         let added_index = random_churn(&mut rng, &network, &mut nodes);
```

Comment:
In the future, since we want to test merging and splitting thoroughly, maybe we should do the churn in 3 stages: adding a lot of nodes (which will generate splits), random churn (like now), dropping almost all nodes (which will test merging). It's probably not in the scope of this PR, though ;)